src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -247,14 +252,12 @@ private void addShutdownListenerToChannel(Channel channel, Boolean withRecovery)
                         errorBuilder.append(channelCause);
                         String errorString = errorBuilder.toString();
                         LOGGER.warn(errorString);
-                        if (withRecovery &&
-                                (errorString.contains("reply-text=PRECONDITION_FAILED")
-                                || errorString.contains("reply-text=NOT_FOUND"))
-                        ) {
+                        if (withRecovery && recoverableErrors.stream().anyMatch(errorString::contains)) {
                             int channelNumber =  channel.getChannelNumber();
                             var pinIdToChannelHolder = getChannelHolderByChannelNumber(channelNumber);
-                            if (pinIdToChannelHolder != null && !pinIdToChannelHolder.getValue().subscribing())
+                            if ((pinIdToChannelHolder != null) && (pinIdToChannelHolder.getValue().state() != ChannelState.SUBSCRIBING)) {
```

Comment:
I don't think that it is okay that we have access to the internal state without a lock. When I suggested using enum instead of two booleans I meant that we should do that inside the holder. All interactions outside should be left as they were. Also, the work with the state should be done under the lock